### DISCUSSION TITLE
Different node mesh sets different colors.

### DISCUSSION THREAD

**Discussion by WdsPjm** [vsg-dev/VulkanSceneGraph#1034] [Q&A]
I have a glb model that contains a cube and a ball. I want the ball to be red and the cube to be green. I want to transfer color to shader through uniform. But how do I know if I'm currently rendering a ball or a cube

**Comment by robertosfield** [comment]
The geometries will either use a material uniform and/or color arrays to provide the existing color.  Try converting the .glb to a .vsgt and then inspect the model to see the layout of the data, this will help you figure out how to traverse the scene and modify the appropriate state.

**Comment by timoore** [comment]
You can either modify the loaded model, as @robertosfield suggested, or you can write a new shader that uses an additional descriptor set for specifying a custom color. There has been some discussion of this lately and changes have been made to GraphicPipelineConfigurator to better support a style of maintaining a hierarchy of descriptor sets. The vsgcustomshaderset example shows the way.
